plugin/src/main/java/io/jenkins/plugins/casc/core/HudsonPrivateSecurityRealmConfigurator.java

```diff
@@ -47,41 +51,111 @@ public HudsonPrivateSecurityRealmConfigurator() {
 
     @CheckForNull
     @Override
-    public CNode describe(HudsonPrivateSecurityRealm instance, ConfigurationContext context) {
+    public CNode describe(HudsonPrivateSecurityRealm instance, ConfigurationContext context)
+        throws Exception {
+        // allow disabling exporting users if an instance has too many
+        if (System.getProperty("io.jenkins.plugins.casc.core.HudsonPrivateSecurityRealmConfigurator.exportUsers", "true").equals("true")) {
+            return super.describe(instance, context);
+        }
         return null;
     }
 
     private static Collection<UserWithPassword> getter(HudsonPrivateSecurityRealm target) {
         return target.getAllUsers().stream()
-                .map(u -> new UserWithPassword(u.getId(), null)) // password isn't actually stored, only hashed
+                .map(u -> {
+                    UserWithPassword user = new UserWithPassword(u.getId(), null);
+                    user.setName(u.getFullName());
+                    user.setDescription(u.getDescription());
+                    List<UserProperty> properties = u.getAllProperties()
+                        .stream()
+                        .filter(userProperty -> !userProperty.getClass().getName().equals("com.cloudbees.plugins.credentials.UserCredentialsProvider$UserCredentialsProperty"))
```

Comment:
Could this be a lambda method reference invocation, so you could add more to the filter?

Reply:
YAGNI, sure when needed 😄

Reply:
it was also for legibility but ack

Reply:
you didn't say that 😉

Reply:
it was implied :laughing: